src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesSynchronisationService.kt

```diff
@@ -20,35 +21,74 @@ class IncentivesSynchronisationService(
       log.debug("received nomis incentive: $this")
 
-      mappingService.findNomisIncentiveMapping(iepEvent.bookingId, iepEvent.iepSeq)?.run {
+      mappingService.findNomisIncentiveMapping(
+        nomisBookingId = bookingId,
+        nomisIncentiveSequence = incentiveSequence
+      )?.let { incentiveMapping ->
         // todo remove this log
-        log.debug("found nomis incentive mapping: $this")
+        log.debug("found nomis incentive mapping: $incentiveMapping")
+        incentiveService.synchroniseUpdateIncentive(
+          iepEvent.bookingId, incentiveMapping.incentiveId,
+          UpdateIncentiveIEP(
+            reviewTime = this.iepDateTime,
+            commentText = this.commentText,
+            current = this.currentIep
+          )
+        )
         telemetryClient.trackEvent(
           "incentive-updated-synchronisation",
           mapOf(
             "bookingId" to iepEvent.bookingId.toString(),
             "incentiveSequence" to iepEvent.iepSeq.toString(),
-            "auditModuleName" to iepEvent.auditModuleName
+            "auditModuleName" to iepEvent.auditModuleName,
+            "currentIep" to this.currentIep.toString()
           ),
           null
         )
+        if (!this.currentIep) {
+          nomisApiService.getCurrentIncentive(iepEvent.bookingId).let { currentIep ->
+            log.debug("updating current IEP $this \nfollowing update to non current IEP: $currentIep")
+            incentiveService.synchroniseUpdateIncentive(
+              iepEvent.bookingId, incentiveMapping.incentiveId,
+              UpdateIncentiveIEP(
+                reviewTime = currentIep.iepDateTime,
+                commentText = currentIep.commentText,
+                current = currentIep.currentIep
+              )
+            )
+            telemetryClient.trackEvent(
+              "incentive-updated-synchronisation",
+              mapOf(
+                "bookingId" to iepEvent.bookingId.toString(),
+                "incentiveSequence" to iepEvent.iepSeq.toString(),
+                "auditModuleName" to iepEvent.auditModuleName,
+                "currentIep" to currentIep.toString()
+              ),
+              null
+            )
+          }
+        }
       } ?: run {
         log.debug("no nomis incentive mapping found")
-        mappingService.createNomisIncentiveSynchronisationMapping(
-          nomisBookingId = iepEvent.bookingId, nomisSequence = iepEvent.iepSeq, incentiveId = 999
-        )
-        telemetryClient.trackEvent(
-          "incentive-created-synchronisation",
-          mapOf(
-            "bookingId" to iepEvent.bookingId.toString(),
-            "incentiveSequence" to iepEvent.iepSeq.toString(),
-            "auditModuleName" to iepEvent.auditModuleName
-          ),
-          null
-        )
+        incentiveService.synchroniseCreateIncentive(this.toIncentive()).also {
+          mappingService.createNomisIncentiveSynchronisationMapping(
```

Comment:
In next PR after my stuff is merged we should switch to calling the migration function - which retries if the create fails. Else we have the possibility that this message will retry creating multiple incentives

Reply:
yes - might be good to talk it through